src/bytecode/mod.rs

```diff
@@ -32,10 +42,44 @@ impl<'a> LuaBytecode {
     /// Generates a fresh register and returns it.
     /// This is used in cases like `x = 1 + 2 + 3` to generate intermmediate
     /// registers in which, for instance, the result of 2 + 3 is stored.
-    pub fn get_new_reg(&mut self) -> String {
+    pub fn get_new_reg(&mut self) -> u64 {
         let reg = self.reg_counter;
+        self.registers.push(Reg::new(reg));
         self.reg_counter += 1;
-        format!("${}", reg).to_string()
+        self.registers.last().unwrap().id()
+    }
+
+    /// Get the register that corresponds to the given identifier.
+    /// If the corresponding register is not found, a new register is created
+    /// and returned.
+    pub fn get_reg(&mut self, id: &str) -> u64 {
+        if let Some(id) = self.reg_map.get(id) {
+            return *id;
+        }
+        let reg_id = self.get_new_reg();
+        self.reg_map.insert(id.to_string().clone(), reg_id);
```

Comment:
You can use the `Entry` API to avoid donig this lokup twice.

Reply:
I tried, but `rust` complains about a mutable and immutable borrow:
```rust
    pub fn get_reg(&mut self, id: &str) -> u64 {
        // self.get_new_reg borrows mutably self, but self.reg_map borrows self immutably.
        *self.reg_map.entry(id.to_string()).or_insert(self.get_new_reg())
    }
```

Reply:
Yes, you need to change the other mention of this above too. It's worth it!

Reply:
What do you mean by the other mention?